### PR TITLE
Support the new VirtualThread states

### PR DESCRIPTION
**Pull request by babsingh**
Related: https://github.com/eclipse-openj9/openj9/issues/18183

New states added:
- TIMED_PARKING = 7
- TIMED_PARKED = 8
- TIMED_PINNED = 9

Old states removed:
- RUNNABLE_SUSPENDED = (RUNNABLE | SUSPENDED)
- PARKED_SUSPENDED = (PARKED | SUSPENDED)

Old states modified:
- YIELDING = 7 -> 10

### CODE REVIEW

**Comment by babsingh** [comment]
This PR depends on https://github.com/ibmruntimes/openj9-openjdk-jdk21/pull/59. This PR and https://github.com/ibmruntimes/openj9-openjdk-jdk21/pull/59 will need to be merged together.

fyi @tajila

**Comment by babsingh** [comment]
@hangshao0 Requesting your review.

**Comment by tajila** [comment]
Are these state changes safe for jdk21? ie. are they JDK22+

**Comment by babsingh** [comment]
> Are these state changes safe for jdk21? ie. are they JDK22+

Yes, it fixes a bug, which also exists in JDK21. See https://bugs.openjdk.org/browse/JDK-8312498 more details.

**Review by hangshao0 — APPROVED** [review]
LGTM.

**Comment by babsingh** [comment]
Personal build: https://hyc-runtimes-jenkins.swg-devops.com/view/OpenJ9%20-%20Personal/job/Pipeline-Build-Test-Personal/19098

Only known failures are seen in the personal build:
- https://github.com/eclipse-openj9/openj9/issues/17851
- https://github.com/eclipse-openj9/openj9/issues/18426
- https://openj9.slack.com/archives/CDS7QE9HB/p1699449481201849

@hangshao0 This PR is ready to be merged.

**Comment by hangshao0** [comment]
Jenkins test sanity.functional,sanity.openjdk aix jdk21 depends ibmruntimes/openj9-openjdk-jdk21#59

**Comment by babsingh** [comment]
> https://openj9-jenkins.osuosl.org/job/Test_openjdk21_j9_sanity.openjdk_ppc64_aix_Personal_testList_0/10/consoleFull

All PR builds have passed; no test failures are seen. There is an infra issue where the scripts cannot end because `Calling Pipeline was cancelled`.

File: runtime/jvmti/jvmtiHelpers.c [hunk 1]
```diff
@@ -855,28 +855,28 @@ getVirtualThreadState(J9VMThread *currentThread, jthread thread)
...
-				break;
+				/* Fall Through */
 			case JVMTI_VTHREAD_STATE_RUNNING:
-				rc = JVMTI_JAVA_LANG_THREAD_STATE_RUNNABLE;
-				break;
+				/* Fall Through */
 			case JVMTI_VTHREAD_STATE_PARKING:
 				/* Fall Through */
+			case JVMTI_VTHREAD_STATE_TIMED_PARKING:
+				/* Fall Through */
```
Comment (keithc-ca) [inline runtime/jvmti/jvmtiHelpers.c:864]: I would recommend against these "fall through" comments when there's no code between case labels. It's even given as a good example in https://github.com/eclipse/omr/blob/master/doc/CodingStandard.md#formatting-switch-statements.

Comment (babsingh) [reply runtime/jvmti/jvmtiHelpers.c:864]: Addressed in https://github.com/eclipse-openj9/openj9/pull/18442.

**Pull request merged**